kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/seek/IndexManager.scala

```diff
@@ -0,0 +1,184 @@
+
+    val s3Path = IndexFilenames.indexFilename(
+      sinkName,
+      topicPartitionOffset.topic.value,
+      topicPartitionOffset.partition,
+      topicPartitionOffset.offset.value
+    )
+
+    logger.debug("[{}] Writing index {} pointing to file {}", sinkName, s3Path, s3PathLocation.path)
+
+    val idxPath = s3PathLocation.root().withPath(s3Path)
+    storageInterface
+      .writeStringToFile(idxPath, s3PathLocation.path)
+      .map(_ => idxPath)
+      .leftMap(
+        ex => {
+          logger.error("[{}] Exception writing index {} pointing to file {}", sinkName, s3Path, s3PathLocation.path, ex)
+          NonFatalS3SinkError(ex.message())
+        }
+      )
+
+  }
+
+  def fallbackOffsetSeek(seeker: LegacyOffsetSeeker, topicPartition: TopicPartition, fileNamingStrategy: S3FileNamingStrategy, bucketAndPrefix: RemoteS3RootLocation): Either[SinkError, Option[TopicPartitionOffset]] = {
+    seeker.seek(topicPartition, fileNamingStrategy, bucketAndPrefix) match {
+      case Left(err) =>
+        logger.error(s"[{}] Error retrieving fallback offset seek topicPartition {} for {}", sinkName, topicPartition, bucketAndPrefix.bucket)
+        err.asLeft
+      case Right(Some((tpo, path))) =>
+        write(tpo, path).map(_ => Some(tpo))
+      case Right(None) =>
+        None.asRight
+    }
+  }
+
+  /**
+   * Seeks the filesystem to find the latyest offsets for a topic/partition.
+   * @param topicPartition the TopicPartition for which to retrieve the offsets
+   * @param fileNamingStrategy the S3FileNamingStrategy to use in the case that a fallback offset seeker is required.
+   * @param bucketAndPrefix the configured RemoteS3RootLocation
+   * @return either a SinkError or an option to a TopicPartitionOffset with the seek result.
+   */
+  def seek(topicPartition: TopicPartition, fileNamingStrategy: S3FileNamingStrategy, bucketAndPrefix: RemoteS3RootLocation): Either[SinkError, Option[TopicPartitionOffset]] = {
+
+    /**
+     * Parses the filename of the index file, converting it to a TopicPartitionOffset
+     * @param maybeIndex option of the index filename
+     * @return either an error, or a TopicPartitionOffset
+     */
+    def indexToOffset(maybeIndex: Option[String]): Either[Throwable, Option[TopicPartitionOffset]] = {
+      maybeIndex match {
+        case Some(index) =>
+          for {
+            offset <- IndexFilenames.offsetFromIndex(index)
+          } yield Some(topicPartition.withOffset(offset))
+        case None => Option.empty[TopicPartitionOffset].asRight
+      }
+    }
+
+    storageInterface.list(bucketAndPrefix.withPath(
+      IndexFilenames.indexForTopicPartition(sinkName, topicPartition.topic.value, topicPartition.partition)
+    ))
+      .leftMap(e => new NonFatalS3SinkError("Couldn't retrieve listing", e.exception))
+      .flatMap(
+        indexes =>
+          if (indexes.size > maxIndexes) {
```

Comment:
if we do this, we would never ever recover without someone manually removing the files right. 
So imagine the connector stops. the first reaction is for the user might be to restart. As it starts, this kicks in and fails.

Should we try to clean first before throwing? what. do you think?

Reply:
Good point.  I was thinking this would require manual intervention, for example fixing permissions on AWS.  However once that manual intervention has been performed there's not any reason to redo the cleanup.  I will fix this.